tests/wicked/2nics/sut/t03_setup_second_card.pm

```diff
@@ -12,29 +12,32 @@
 #             Jose Lausuch <jalausuch@suse.com>
 #             Clemens Famulla-Conrad <cfamullaconrad@suse.de>
 
-use base 'wickedbase';
-use strict;
-use warnings;
+use Mojo::Base 'wickedbase';
 use testapi;
-use utils 'file_content_replace';
+use lockapi;
+use utils qw(file_content_replace systemctl);
 use network_utils 'iface';
 
 
 sub run {
     my ($self, $ctx) = @_;
-    my $config  = '/etc/sysconfig/network/ifcfg-' . $ctx->iface();
-    my $config2 = '/etc/sysconfig/network/ifcfg-' . $ctx->iface2();
+    mutex_wait('t03_dhcpd_setup_complete');
+    my $cfg_ifc1 = '/etc/sysconfig/network/ifcfg-' . $ctx->iface();
+    my $cfg_ifc2 = '/etc/sysconfig/network/ifcfg-' . $ctx->iface2();
     record_info('Info', 'Set up a second card');
-    $self->get_from_data('wicked/static_address/ifcfg-eth0_second_card', $config);
-    $self->get_from_data('wicked/dynamic_address/ifcfg-eth0',            $config2);
+    $self->get_from_data('wicked/dynamic_address/ifcfg-eth0', $cfg_ifc1);
+    $self->get_from_data('wicked/static_address/ifcfg-eth0',  $cfg_ifc2);
+    $self->wicked_command('ifdown', 'all');
+    systemctl('restart wicked');
     $self->wicked_command('ifup', $ctx->iface());
     $self->wicked_command('ifup', $ctx->iface2());
-    my $iface_ip  = $self->get_ip(type => 'second_card');
-    my $iface_ip2 = $self->get_ip(type => 'dhcp');
+    my $iface_ip  = $self->get_ip(type => 'dhcp_2nic');
+    my $iface_ip2 = $self->get_ip(type => 'host');
     validate_script_output('ip a s dev ' . $ctx->iface(),  sub { /$iface_ip/ });
     validate_script_output('ip a s dev ' . $ctx->iface2(), sub { /$iface_ip2/ });
-    die "unable to ping " . $ctx->iface()  unless $self->ping_with_timeout(type => 'host',        interface => $ctx->iface());
-    die "unable to ping " . $ctx->iface2() unless $self->ping_with_timeout(type => 'second_card', interface => $ctx->iface2());
+    die "unable to ping " . $ctx->iface()  unless $self->ping_with_timeout(type => 'host',        interface => $ctx->iface2());
```

Comment:
error message would say about iface while you passing iface2 , looks like you wanted to just have this , right? : 

```
die "unable to ping " . $ctx->iface()  unless $self->ping_with_timeout(type => 'second_card',        interface => $ctx->iface());
die "unable to ping " . $ctx->iface2() unless $self->ping_with_timeout(type => 'host', interface => $ctx->iface2());
```

Reply:
jap